lib/libvfio-user.c

```diff
@@ -713,51 +691,73 @@ handle_dma_map(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
                                     dma_map->size, fd, dma_map->offset,
                                     prot);
     if (ret < 0) {
-        ret = errno;
-        vfu_log(vfu_ctx, LOG_ERR, "failed to add DMA region %s: %m", rstr);
+        vfu_log(vfu_ctx, LOG_ERR,
+                "failed to add DMA region [%#lx, %#lx) offset=%#lx flags=%#x: %m",
+                dma_map->addr, dma_map->addr + dma_map->size, dma_map->offset,
+                dma_map->flags);
         if (fd != -1) {
             close(fd);
         }
-        return ERROR_INT(ret);
+        return ret;
     }
-
+    idx = ret;
+    ret = 0;
     if (vfu_ctx->dma_register != NULL) {
-        vfu_ctx->dma_register(vfu_ctx, &vfu_ctx->dma->regions[ret].info);
+        vfu_ctx->dma_register(vfu_ctx, &vfu_ctx->dma->regions[idx].info);
     }
-    return 0;
+    vfu_log(vfu_ctx, LOG_DEBUG,
+            "added DMA region [%#lx, %#lx) offset=%#lx flags=%#x",
+            dma_map->addr, dma_map->addr + dma_map->size, dma_map->offset,
+            dma_map->flags);
+    return ret;
 }
 
 int
-handle_dma_unmap(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
-                 struct vfio_user_dma_unmap *dma_unmap)
+handle_dma_map(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
+               struct vfio_user_dma_map *dma_map)
 {
-    size_t out_size;
-    int ret = 0;
-    bool unmap_all = false;
-    char rstr[1024];
+    int ret;
 
     assert(vfu_ctx != NULL);
     assert(msg != NULL);
-    assert(dma_unmap != NULL);
+    assert(dma_map != NULL);
 
-    if (msg->in_size < sizeof(*dma_unmap) || dma_unmap->argsz < sizeof(*dma_unmap)) {
-        vfu_log(vfu_ctx, LOG_ERR, "bad DMA unmap region size=%zu argsz=%u",
-                msg->in_size, dma_unmap->argsz);
+    if (msg->in_size < sizeof(*dma_map) || dma_map->argsz < sizeof(*dma_map)) {
+        vfu_log(vfu_ctx, LOG_ERR, "bad DMA map region size=%zu argsz=%u",
+                msg->in_size, dma_map->argsz);
         return ERROR_INT(EINVAL);
     }
 
-    if ((dma_unmap->flags & VFIO_DMA_UNMAP_FLAG_GET_DIRTY_BITMAP) &&
-            (dma_unmap->flags & VFIO_DMA_UNMAP_FLAG_ALL)) {
-        vfu_log(vfu_ctx, LOG_ERR, "invalid DMA flags=%#x", dma_unmap->flags);
-        return ERROR_INT(EINVAL);
-    }
+    vfu_log(vfu_ctx, LOG_DEBUG,
+            "adding DMA region [%#lx, %#lx) offset=%#lx flags=%#x",
+            dma_map->addr, dma_map->addr + dma_map->size, dma_map->offset,
+            dma_map->flags);
 
-    snprintf(rstr, sizeof(rstr), "[%#lx, %#lx) flags=%#x",
-             dma_unmap->addr, dma_unmap->addr + dma_unmap->size, dma_unmap->flags);
+    if (vfu_ctx->quiesce != NULL) {
+        vfu_log(vfu_ctx, LOG_DEBUG, "quiescing device");
+        ret = vfu_ctx->quiesce(vfu_ctx);
+        if (ret < 0) {
+            if (errno == EBUSY) {
+                vfu_log(vfu_ctx, LOG_DEBUG, "device will quiesce asynchronously");
```

Comment:
why do we get debug for this but not the other quiesce users.

Reply:
No good reason, I think I was debugging something. Shall I add this to all others? Or remove it from everywhere?

Reply:
can we commonize at all?